keras_core/layers/preprocessing/random_crop.py

```diff
@@ -52,41 +52,121 @@ class RandomCrop(Layer):
             `name` and `dtype`.
     """
 
-    def __init__(self, height, width, seed=None, name=None, **kwargs):
-        if not tf.available:
-            raise ImportError(
-                "Layer RandomCrop requires TensorFlow. "
-                "Install it via `pip install tensorflow`."
-            )
-
+    def __init__(
+        self, height, width, seed=None, data_format=None, name=None, **kwargs
+    ):
         super().__init__(name=name, **kwargs)
+        self.height = height
+        self.width = width
         self.seed = seed or backend.random.make_default_seed()
-        self.layer = tf.keras.layers.RandomCrop(
-            height=height,
-            width=width,
-            seed=self.seed,
-            name=name,
-        )
+        self.seed_generator = SeedGenerator(seed)
+        self.data_format = backend.standardize_data_format(data_format)
+
+        if self.data_format == "channels_first":
+            self.heigh_axis = -2
+            self.width_axis = -1
+        elif self.data_format == "channels_last":
+            self.height_axis = -3
+            self.width_axis = -2
+
         self.supports_masking = False
         self.supports_jit = False
         self._convert_input_args = False
         self._allow_non_tensor_positional_args = True
 
     def call(self, inputs, training=True):
-        if not isinstance(inputs, (tf.Tensor, np.ndarray, list, tuple)):
-            inputs = tf.convert_to_tensor(backend.convert_to_numpy(inputs))
-        outputs = self.layer.call(inputs, training=training)
-        if (
-            backend.backend() != "tensorflow"
-            and not backend_utils.in_tf_graph()
-        ):
-            outputs = backend.convert_to_tensor(outputs)
+        inputs = self.backend.cast(inputs, self.compute_dtype)
+        input_shape = self.backend.shape(inputs)
+        is_batched = len(input_shape) > 3
+        inputs = (
+            self.backend.numpy.expand_dims(inputs, axis=0)
+            if not is_batched
+            else inputs
+        )
+
+        h_diff = input_shape[self.height_axis] - self.height
+        w_diff = input_shape[self.width_axis] - self.width
+
+        def random_crop():
```

Comment:
To avoid an eager performance overhead, please move this to a method, `_randomly_crop_inputs` (otherwise redefining the function at each execution is slow)